lib/plugins/plugin-validator.js

```diff
@@ -21,11 +21,21 @@ const knownKeys = [
   'meta'
 ]
 
+/**
+ * @param {string} executionPath
+ * @param {string} pathToValidate
+ * @param {string} key
+ */
 function checkPathExists (executionPath, pathToValidate, key) {
   const absolutePathToValidate = path.join(executionPath, pathToValidate)
   if (!fse.existsSync(absolutePathToValidate)) errors.push(`In section ${key}, the path '${pathToValidate}' does not exist`)
 }
 
+/**
+ * @param {string} executionPath
+ * @param {string} nunjucksFileName
+ * @param {string | string[]} nunjucksPaths
```

Comment:
I can't see any logic to support `nunjucksPath` being a string rather than an array of strings – I think we'd end up iterating over each character in the string on line 48?

It looks like this might also be optional?

Should it be…

```suggestion
 * @param {string[]} [nunjucksPaths]
```

Reply:
Definitely give https://github.com/alphagov/govuk-prototype-kit/pull/2354 a try to see lots more red squiggles

Similar to 77b794393a21b7616067ccb7da6646da3c70496a you'll see it's still possible to access `configEntry.path` and `configEntry.importFrom` even though `configEntry` might be a string etc

Or sometimes we just check the `key` name but not the type:

https://github.com/alphagov/govuk-prototype-kit/blob/bb9528b488e662cd5b6a529e761b9c3714263348/lib/plugins/plugin-validator.js#L188-L190

Reply:
Looks like possibly the kit supports `nunjucksPaths` being a string but this validator doesn't allow for it?

I mean, technically you _can_ pass a string, but it'll try and iterate over it character by character and result in errors – what are we trying to document in the JSDoc? Technical compatibility or how it's designed to be used?

Reply:
>Looks like possibly the kit supports nunjucksPaths being a string but this validator doesn't allow for it?

Yeah that's it exactly

Just tried it with GOV.UK and everything works. The code [was simplified recently](https://github.com/alphagov/govuk-prototype-kit/commit/bb1cebbd5c689942da15d5bc658487e828d97ce8) but it's `flat()` that does the magic:

```js
nunjucksViews.push(...[config.nunjucksPaths].flat()
  .map(nunjucksPath => path.join(baseDir, nunjucksPath))
```

Reply:
>what are we trying to document in the JSDoc? Technical compatibility or how it's designed to be used?

I added JSDoc to document how the kit is _currently_ used

That way (in theory) the compiler can error or show red squiggles on all the code that doesn't support the current use

Except for looping strings, which is valid 🤦‍♂️

Reply:
I've pushed up a tweak to handle `undefined` and strings at the same time in https://github.com/alphagov/govuk-prototype-kit/pull/2386/commits/5e534e78b1b71c666f3da66a71b55c47039333f2

Ideally we should leave broken things broken and triage them?